src/python/pants/base/payload_field.py

```diff
@@ -139,8 +139,12 @@ def _compute_fingerprint(self):
     hasher.update(self._rel_path)
     for source in sorted(self.relative_to_buildroot()):
       hasher.update(source)
-      with open(os.path.join(get_buildroot(), source), 'rb') as f:
-        hasher.update(f.read())
+      abs_path = os.path.join(get_buildroot(), source)
+      if not os.path.exists(abs_path):
+        hasher.update('\a\n\n\t[FILE DOES NOT EXIST]\t\n\n\a')
+      else:
```

Comment:
in prepare() add:
```
round_manager.require_data('deferred_sources')
```

Now somehow we get a DeferredSourcesField() into place. I think that is by making `sources = from_target('...')` You can add an integration test for importing from wire in this way - there is already a protobuf test that way you could copy: `examples/src/java/org/pantsbuild/example/protobuf/unpacked_jars/`